src/core_modules/capture-core/components/Pages/MainPage/WorkingListsCommon/hooks/useWorkingListsCommonStateManagement.js

```diff
@@ -25,103 +25,151 @@ import {
 
     const nextFilters = useSelector(({ workingListsMeta }) =>
-        workingListsMeta[listId] && workingListsMeta[listId].next && workingListsMeta[listId].next.filters);
+        workingListsMeta[storeId] && workingListsMeta[storeId].next && workingListsMeta[storeId].next.filters);
     const filtersState = useMemo(() => ({ ...appliedFilters, ...nextFilters }), [
         appliedFilters,
         nextFilters,
     ]);
 
     const viewDispatch = useMemo(() => ({
-        onLoadView: workingListDispatch(initListView,
-            (selectedTemplate: Object, context: Object, meta: Object) => [
+        onLoadView: (selectedTemplate: Object, context: Object, meta: Object) =>
+            dispatch(initListView(
                 selectedTemplate,
                 context, {
                     ...meta,
                     categoryCombinationMeta,
-                    listId,
+                    storeId,
+                    workingListsType,
                 },
-            ]),
-        onUpdateList: workingListDispatch(updateList,
-            (queryArgs: Object, columnsMetaForDataFetching: Object) => [
+            )),
+        onUpdateList: (queryArgs: Object, columnsMetaForDataFetching: Object) =>
+            dispatch(updateList(
                 queryArgs, {
                     columnsMetaForDataFetching,
                     categoryCombinationMeta,
-                    listId,
+                    storeId,
+                    workingListsType,
                 },
-            ]),
-        onCancelLoadView: workingListDispatch(initListViewCancel),
-        onCancelUpdateList: workingListDispatch(updateListCancel),
-        onSortList: workingListDispatch(sortList),
-        onSetListColumnOrder: workingListDispatch(setListColumnOrder),
-        onUpdateFilter: workingListDispatch(setFilter),
-        onClearFilter: workingListDispatch(clearFilter),
-        onSelectRestMenuItem: workingListDispatch(restMenuItemSelected),
-        onChangePage: workingListDispatch(changePage),
-        onChangeRowsPerPage: workingListDispatch(changeRowsPerPage),
-    }), [listId, workingListDispatch, categoryCombinationMeta]);
+            )),
+        onCancelLoadView: (...args) => dispatch(initListViewCancel(...args, storeId)),
+        onCancelUpdateList: (...args) => dispatch(updateListCancel(...args, storeId)),
+        onSortList: (...args) => dispatch(sortList(...args, storeId)),
+        onSetListColumnOrder: (...args) => dispatch(setListColumnOrder(...args, storeId)),
+        onUpdateFilter: (...args) => dispatch(setFilter(...args, storeId)),
+        onClearFilter: (...args) => dispatch(clearFilter(...args, storeId)),
+        onSelectRestMenuItem: (...args) => dispatch(restMenuItemSelected(...args, storeId)),
+        onChangePage: (...args) => dispatch(changePage(...args, storeId)),
+        onChangeRowsPerPage: (...args) => dispatch(changeRowsPerPage(...args, storeId)),
```

Comment:
I dont know your opinion on this but I find it always better when I can see what is being passed around. 

`...args` take this away a little bit now. I mean now is a little bit tricky to know what is hidden inside the args. To be fair I could derive it from looking the signature of the action creator.

But I am thinking to myself that since the arguments we are hiding here using the `...args` technique are maximum 2 values would it be better to explicitly define what is in the args and leave the mysteries aside?

Reply:
fair.

Reply:
Yep I was just thinking out loud mostly. I dont have a strong opinion on this.

> One pro is that it makes it really easy to see the changes to the argument list

Can you explain that to me a little bit more?